tip/environment.py

```diff
@@ -0,0 +1,44 @@
+import os
+import json
+
+from tip import config
+from tip.util import parse_package_specifier
+
+
+class Environment:
+    """A TIP environment."""
+
+    def __init__(self, packages, path):
```

Comment:
It's not obvious what is `Environment({}, Environment.locate(...))` without reading the context.
Creating an empty Environment seems like a common use-case, so it's not good to require for a common use-case to specify empty dict of packages. Also, if in the future you will add some new params, then to make the interface compatible you will have to add new params after `path` param which looks not pretty.
```suggestion
    def __init__(self, *, path, packages=None):
```